packages/nextra/src/use-internals.ts

```diff
@@ -15,19 +15,23 @@ export function useInternals() {
   const { route } = useRouter()
   const rerender = useState({})[1]
 
-  useEffect(() => {
-    if (IS_PRODUCTION) return
-    const trigger = () => rerender({})
+  // The HMR handling logic is not needed for production builds, the condition
+  // should be removed after compilation and it's fine to put the effect under
+  // if, because hooks' order is still stable.
+  if (process.env.NODE_ENV !== 'production') {
+    useEffect(() => {
```

Comment:
you can use `eslint-disable-next-line` comment for this case since this code will be tree shaked